gridt/src/app/core/settings.service.ts

```diff
@@ -1,107 +1,136 @@
-    private http: HttpClient,
-    private auth: AuthService,
-    private secStore: SecureStorageService
-  ) {}
-
-  public error_codes = {}
-
-  /**
-   * Create ReplaySubject which yields the latest user settings.
-   */
-  private _user_settings$ = new ReplaySubject<Settings>(1);
-  /*
-   * The depth of the ReplaySubject is set to 1, which makes sure only the latest
-   * available settings are retured to the user.
-   */
-
-  /**
-   * Transforms the user settings Subject to an Observable
-   */
-  get the_user_settings$ (): Observable<Settings> {
-    return this._user_settings$.asObservable();
-  }
-
-  /**
-   * Observable to obtain settings from secure storage
-   */
-  private getLocalSettings = new Observable<Settings>( (observer) => {
-    observer.next({username: "John1"});
-
-    let set = this.setter.subscribe(set => {
-      console.log(set.username);
-      observer.next({username: set.username});
-    });
-
-    observer.complete();
-  });
-
-  private setter: Observable<Settings> = forkJoin({
-    username: this.secStore.get$("username")
-  })
-
-  /**
-   * Store settings in the secure storage.
-   * @param settings Settings to be stored.
-   */
-  private storeLocalSettings(settings: Settings): void {
-    this.secStore.set$("username", settings).subscribe();
-  }
-
-  /**
-   * Update local storage whenever the settings change.
-   */
-  // _user_settings$.subscribe(
-  //   (new_settings) => {
-  //     this.storeLocalSettings(new_settings);
-  //     console.log("New settings stored in secure local storage.")
-  //   }
-  // )
-
-  public populateStorage(): void {
-    console.log("Populating secure storage.")
-    this.secStore.set$("username", "John2").subscribe();
-  }
-
-  /**
-   * Observable to obtain settings from server
-   */
-  private getServerSettings = new Observable<Settings>( (observer) => {
-    observer.next();
-    observer.complete();
-  })
-
-  /**
-   * Combine Local and server settings
-   */
-  public getUserSettings(): void {
-
-    forkJoin({
-      local: this.getLocalSettings,
-      server: this.getServerSettings
-    }).pipe(
-      map( (settings) => {
-        return {...settings.local, ...settings.server}
-        // Server settings have priority over local settings
-      })
-    ).subscribe( (set) => this._user_settings$.next(set) );
-  }
-}
+import { Injectable } from "@angular/core";
+import { Observable, ReplaySubject, BehaviorSubject, throwError, forkJoin, of } from "rxjs";
+import { HttpClient } from "@angular/common/http";
+import { map, tap, flatMap, pluck, delay, catchError } from "rxjs/operators";
+import { Identity } from './identity.model';
+import { Settings } from './settings.model';
+import { AuthService } from './auth.service';
+import { SecureStorageService } from './secure-storage.service';
+
+@Injectable({
+  providedIn: "root"
+})
+export class SettingsService {
+  /*
+   * Subscribe to this observable to ready the API.
+   */
+  public URL = "https://api.gridt.org";

```

Comment:
I think it would be logical to have this in the settings as well. But if we would make settings server specific this setting would be paradoxical.

Reply:
This becomes redundant in settings.service if we move the getServerSettings function to the api.service. In the settings service a function should be added which gets the preferred server URL from local storage. This function can then be called from the api.service.

Reply:
Getting the setting "which server should I address" from a server sounds a little strange to me, but I agree that once local storage is functional, this should be fixed.